### ISSUE TITLE
Forcibly re-run a migration

### ISSUE THREAD

**Issue by anwar-gazi** [cakephp/phinx#717]
I executed a migration which gone wrong, so next time I fixed the code inside that migration and then tried the php -f <path to phinx bin> migration command but this time that migration didn't run

**Comment by robmorgan** [comment]
@minhaj-resgef there is not currently a force command, although I have plans for one.

**Comment by minhaj-resgef** [comment]
hi dev, thanks,
is there any way to manually clear that cache?

**Comment by robmorgan** [comment]
@minhaj-resgef yes, open your database. Find the `phinxlog` table and deleted the specified row that represents your migration.,

**Comment by minhaj-resgef** [comment]
I've written a php script to do that automatically. And further, added a -f switch to delete migration history and run again.
I want to share the code.

**Comment by igorsantos07** [comment]
The best way I have to try out migrations while developing is this:
1. run migration, but only part of it works
2. comment out what reverts in `down()` what has not run
3. `migrate rollback`
4. fix `up()`, uncomment pieces of `down()`
5. `migrate`
6. did not work? repeat.

You basically revert the half-migration using its own `down()` until you get it right. This way you do not have to deal with half-changes in the database or `phinxlog` table by hand.

**Comment by dereuromark** [comment]
I dont think given the many side effects that a better solution can be found here.
The given workaround is the best approach.

**Comment by robmorgan** [comment]
@dereuromark thanks Mark for resolving this after 10 years ;)
